### PR TITLE
Auto implicit fixes

### PR DESCRIPTION
**Pull request by FSMaxB**
Continuing #921

### CODE REVIEW

**Comment by josdejong** [comment]
Thanks Max, I will review your code at the end of this week. Looks good at first sight.

**Comment by FSMaxB** [comment]
You should probably review both commits separately, since the first one mostly changes indentation, thereby bloating the diff.

**Comment by josdejong** [comment]
ok makes sense

**Comment by josdejong** [comment]
Your code looks good Max, this is indeed the behavior we're looking for. Thanks

There is one thing in the code that I don't fully understand, function `calculateNecessaryParentheses` now has a new argument `implicit`, but as far as I can see this argument is not used in the function?

File: lib/expression/node/OperatorNode.js [hunk 1]
```diff
@@ -158,154 +158,184 @@ function factory (type, config, load, typed) {
...
+            return false;
+          });
+        }
+        break;
+    }
+
+    //handles an edge case of 'auto' parentheses with implicit multiplication of ConstantNode
+    //In that case print parentheses for ParenthesisNodes even though they normally wouldn't be
+    //printed.
+    if ((args.length >= 2) && (root.getIdentifier() === 'OperatorNode:multiply') && root.implicit && (parenthesis === 'auto') && (implicit === 'hide')) {
```
Comment (FSMaxB) [inline lib/expression/node/OperatorNode.js]: @josdejong: The new parameter is used here!

`root.implicit` says that the current multiplication is implicit. `implicit` is the option that has been passed in. Either `hide` or `show`.

Comment (josdejong) [inline lib/expression/node/OperatorNode.js]: ahhh, I simply overlooked it.

**Pull request merged**